### ISSUE TITLE
M 2.4.3-p1 - TypeError in AdminSessionsManager.php:338 (strtotime() expects parameter 1 to be string, int given)

### ISSUE THREAD

**Issue by simonmaass** [magento/magento2#34415]
### Preconditions (*)

1. Magento 2.4.3-p1

### Steps to reproduce (*)

1. When I try to login with SSO...

### Expected result (*)

1. Login works

### Actual result (*)

```
TypeError: strtotime() expects parameter 1 to be string, int given
#28 /var/www/share/meevo.io/releases/66/vendor/magento/module-security/Model/AdminSessionsManager.php(338): strtotime
```

magento2/app/code/Magento/Security/Model/AdminSessionsManager.php 

```
private function lastProlongIsOldEnough()
    {
        $lastProlongTimestamp = strtotime($this->getCurrentSession()->getUpdatedAt());
        $nowTimestamp = $this->authSession->getUpdatedAt();

        $diff = $nowTimestamp - $lastProlongTimestamp;

        return (float) $diff > $this->getIntervalBetweenConsecutiveProlongs();
    }

```

In this file the line https://github.com/magento/magento2/compare/2.4.3...2.4.3-p1#diff-24fbc4a93c8c25f9ee036f3bf353c3dfca22ee7ff35d0797b6d1c1d7dc012386R6

declare(strict_types=1);

$this->getCurrentSession()->getUpdatedAt() returns int as timestamp... for example 1635087142

Here is the complete trace

```
TypeError: strtotime() expects parameter 1 to be string, int given
#28 /vendor/magento/module-security/Model/AdminSessionsManager.php(338): strtotime
#27 /vendor/magento/module-security/Model/AdminSessionsManager.php(338): Magento\Security\Model\AdminSessionsManager::lastProlongIsOldEnough
#26 /vendor/magento/module-security/Model/AdminSessionsManager.php(137): Magento\Security\Model\AdminSessionsManager::processProlong
#25 /vendor/magento/module-security/Model/Plugin/AuthSession.php(85): Magento\Security\Model\Plugin\AuthSession::aroundProlong
#24 /vendor/magento/framework/Interception/Interceptor.php(135): Magento\Backend\Model\Auth\Session\Interceptor::Magento\Framework\Interception\{closure}
#23 /vendor/magento/framework/Interception/Interceptor.php(153): Magento\Backend\Model\Auth\Session\Interceptor::___callPlugins
#22 /Magento/Backend/Model/Auth/Session/Interceptor.php(23): Magento\Backend\Model\Auth\Session\Interceptor::prolong
#21 /vendor/magento/module-backend/App/Action/Plugin/Authentication.php(127): Magento\Backend\App\Action\Plugin\Authentication::aroundDispatch
#20 /vendor/magento/framework/Interception/Interceptor.php(135): Magento\Backend\Controller\Adminhtml\Auth\Login\Interceptor::Magento\Framework\Interception\{closure}
#19 /vendor/magento/framework/Interception/Interceptor.php(153): Magento\Backend\Controller\Adminhtml\Auth\Login\Interceptor::___callPlugins
#18 /Magento/Backend/Controller/Adminhtml/Auth/Login/Interceptor.php(32): Magento\Backend\Controller\Adminhtml\Auth\Login\Interceptor::dispatch
#17 /vendor/magento/framework/App/FrontController.php(245): Magento\Framework\App\FrontController::getActionResponse
#16 /vendor/magento/framework/App/FrontController.php(212): Magento\Framework\App\FrontController::processRequest
#15 /vendor/magento/framework/App/FrontController.php(147): Magento\Framework\App\FrontController::dispatch
#14 /vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\App\FrontController\Interceptor::___callParent
#13 /vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\App\FrontController\Interceptor::Magento\Framework\Interception\{closure}
#12 /src/Infortis/ext-magento2-cgen/src/Plugin/Magento/Framework/App/FrontController.php(32): Infortis\Cgen\Plugin\Magento\Framework\App\FrontController::aroundDispatch
#11 /vendor/magento/framework/Interception/Interceptor.php(135): Magento\Framework\App\FrontController\Interceptor::Magento\Framework\Interception\{closure}
#10 /vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\App\FrontController\Interceptor::___callPlugins
#9 /Magento/Framework/App/FrontController/Interceptor.php(23): Magento\Framework\App\FrontController\Interceptor::dispatch
#8 /vendor/magento/framework/App/Http.php(116): Magento\Framework\App\Http::launch
#7 /vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\App\Http\Interceptor::___callParent
#6 /vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\App\Http\Interceptor::Magento\Framework\Interception\{closure}
#5 /vendor/justbetter/magento2-sentry/Plugin/GlobalExceptionCatcher.php(58): JustBetter\Sentry\Plugin\GlobalExceptionCatcher::aroundLaunch
#4 /vendor/magento/framework/Interception/Interceptor.php(135): Magento\Framework\App\Http\Interceptor::Magento\Framework\Interception\{closure}
#3 /vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\App\Http\Interceptor::___callPlugins
#2 /Magento/Framework/App/Http/Interceptor.php(23): Magento\Framework\App\Http\Interceptor::launch
#1 /vendor/magento/framework/App/Bootstrap.php(264): Magento\Framework\App\Bootstrap::run
#0 /index.php(29): null

```
---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [x] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by kanhaiya5590** [comment]
Very similar problem issue found in Magento v2.4.3-p1 or can say same issue what @hostep found in Magento v2.3.7-p2
 
 `main.CRITICAL: TypeError: strtotime() expects parameter 1 to be string, null given in /home/developer/UAT-GAF/vendor/magento/module-security/Model/AdminSessionInfo.php:136`
 
 This not allow to access admin panel. and in PRODUCTION mode getting **An error has happened during application run. See exception log for details.**

**Comment by kanhaiya5590** [comment]
> Very similar problem/issue found in Magento v2.4.3-p1 or can say same issue what @hostep found in Magento v2.3.7-p2
> 
> `main.CRITICAL: TypeError: strtotime() expects parameter 1 to be string, null given in /home/web/magento/vendor/magento/module-security/Model/AdminSessionInfo.php:136`
> 
> This not allow to access admin panel. and in PRODUCTION mode getting **An error has happened during application run. See exception log for details.**

**Comment by simonmaass** [comment]
@kanhaiya5590 I dont think your patch fixes this error in total in AdminSessionsManager...

As far as I can see `$this->getCurrentSession()->getUpdatedAt()` returns an int if not null.. 
`TypeError: strtotime() expects parameter 1 to be string, int given`

**Comment by ihor-sviziev** [comment]
@kanhaiya5590, could you create a pull request?

**Comment by kanhaiya5590** [comment]
> @kanhaiya5590 I dont think your patch fixes this error in total in AdminSessionsManager...
> 
> As far as I can see `$this->getCurrentSession()->getUpdatedAt()` returns an int if not null.. `TypeError: strtotime() expects parameter 1 to be string, int given`

Thanks as i was got issue for null but *i updated the patch file* with an extra condition check` is_numeric($lastUpdatedTime)`, hope this will solve your concern.

Please do share the feedback.

**Comment by ihor-sviziev** [comment]
@sidolov FYI I set priority to P1 as this issue prevents logging in to the admin

**Comment by jgmdigital** [comment]
Is there any update on this issue?

**Comment by ihor-sviziev** [comment]
@jgmdigital,
As for now, you can apply the changes from the following PR: https://github.com/magento/magento2/pull/34514

**Comment by kanevbg** [comment]
Magento version 2.3.7-p2 has the same issue.

**Comment by kanevbg** [comment]
@ihor-sviziev I have patched it back then by doing minimalist patch looking at the commit following similar principle (only changed things which matter to the code/logic), and I have applied it with hotfix type of patching in the project. Here is the patch file I use for 2.3.7-p2, in case someone finds it useful:
[HD-MDVA-42269-1.zip](https://github.com/magento/magento2/files/8816103/HD-MDVA-42269-1.zip)

**Comment by tuyennn** [comment]
Confirm same issue on EE 2.4.3-p2

**Comment by christianbookpwood** [comment]
Just FYI, this is also an issue in 2.3.7-p4.

Does anyone know if/when this issue has been fixed in Magento's core code?

**Comment by hostep** [comment]
@christianbookpwood: The code from [the PR that fixed this](https://github.com/magento/magento2/pull/34514) is included in Magento 2.4.5 and higher.
